### ISSUE TITLE
Summary page

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#352]
There are certain bits of information that are important to most lunar programs, which have to be pulled from different parts of your space program and might be helpful to have in one place for quick access.  This could be handy when you're loading a save game or just playing a game quickly (or after a few brews).  I'm thinking of a page that would display some or all of these:
Duration Level accomplished (None, A-F)
DM in Orbit (Turn turn, This turn and next, Expected next turn)
DM Safety
Kicker Safety (e.g. 0% / 23% or 84% / 0%)
LM Safety (e.g., 0% / 35% or 52% / 0%)
LM Points
EVA Safety
Lunar Recon

I don't know where it could go, though.  Possibly it could be its own building: we could add a building to each spaceport that would bring it up.  That's possibly a nontrivial change though.  In any case, adding this sort of a page somewhere would allow me to take DUR LVL out of the Satellite Building; I added it there in version .4.8.  It doesn't really belong there since it's unrelated to docking; I just couldn't think of a better place to put it.  It would be nice to put DOCKING back up at the top where it should be.
![SatelliteBldg](https://user-images.githubusercontent.com/3118014/80007059-99d57080-847a-11ea-8f4b-e9706d8409a3.png)

This would have the

**Comment by rnyoakum** [comment]
In my mind, the most logical options are either
- A new building, or
- The Museum of Space History

Using a new building would require
- A building graphic,
- deciding where to place it,
- creating a name for it

Adding a new entry to the Museum of Space History would be a bit easier
- an icon in the Museum's menu, and
- creating a name for it

Either one would require a new interface screen. The hardest part of that would be creating the visual design so the information is displayed attractively. RIS likes tossing a graphic or two into every page. In terms of coding, it would be a pretty standard setup, since there wouldn't be much need for interaction.

**Comment by peyre** [comment]
I'm glad this doesn't look like it would be prohibitive to do.  I'll have to kick around design ideas a little bit, though design isn't my strong suit.

**Comment by peyre** [comment]
What if it looked something like this?  (And by "like this" I don't mean "exactly this", like some people I could name; this is just sort of concept art.)
![Progress Summary](https://user-images.githubusercontent.com/3118014/80030109-9c48c200-849c-11ea-9f8f-6d5f7b81d43f.png)

And for a button, we could maybe do something this this.  It's supposed to be a chalkboard where they're outlining the steps needed to do the job, like in Apollo 13 or "From the Earth to the Moon".
![Progress](https://user-images.githubusercontent.com/3118014/80030330-e7fb6b80-849c-11ea-8297-3092a156e77a.png)

**Comment by rnyoakum** [comment]
That looks pretty good, and I like the little icon.

**Comment by peyre** [comment]
Thanks!  Did you notice the detail that the Cricket shows as the pre-prototype image because it hasn't been started yet?

**Comment by rnyoakum** [comment]
> Thanks! Did you notice the detail that the Cricket shows as the pre-prototype image because it hasn't been started yet?

I _hadn't_ noticed the prototype vs implemented image design, which is really cool.

> Of course, those images above aren't full of the splashy colors RIS likes to use; think of them as sketches or storyboards or whatever.

I have a thought on that: the program ratings could be color-coded based on (roughly) how safe they are deemed to be. This could be based on a general x%-y% is comfortable, y%-z% is cautious but okay, below z% is risky, or it could be based on an approach of green=Max Safety, yellow = Max R&D, white = room to research. Do you think either of those is a good fit?

> OH! And, maybe the Soviet player could should show the C Kicker just after the B-Kicker.

Definitely.

> And by "like this" I don't mean "exactly this", like some people I could name; this is just sort of concept art

Err... I guess I've been a little too literal in my design reviews in the past? I'll try to ease up on that.

**Comment by peyre** [comment]
No no, the "like some people I could name" didn't mean you; I was thinking of a boss years ago, who asked me to create a page "something like that" and later it turned out he meant "JUST like this".  Old memories.

Yes, color coding by Safety is a good and logical idea.  And of course, headings should probably be boldfaced.  Being not too creative myself, I came up with a pretty much no-nonsense page full of gray, and figured a final version of it would have more of RIS's happy colors in it.

Thanks, I think showing drawing-board images for things that haven't been prototyped would be kind of important.

**Comment by peyre** [comment]
Another thing that might be handy to have is a checklist of milestones and duration levels, which would make it clear what you still have left to accomplish on your way to the Moon.

**Comment by peyre** [comment]
I'm not sure I'm entirely comfortable with that.  It shows the milestones needed, in order, but it doesn't seem to display requirements very well.  Maybe the "Penalties for Skipping Steps" table in the Player's Aid Card would be a better template for this sort of screen.

**Comment by peyre** [comment]
Also I think in the "concept art" above for the Summary Page, it might be good to include the number of spacewalks performed successfully.  That might help in keeping track of where you are in the game.

**Comment by peyre** [comment]
Another idea for this page is a button that would pop up a summary of what missions you've accomplished so far, e.g.:

ORBITAL SATELLITE (X2)
SUBORBITAL
ORBITAL
ORBITAL EVA DURATION (B)
LUNAR FLYBY (X3)
ORBITAL EVA DURATION (C)
VENUS FLYBY

Though this might get tricky - do we include things like Unmanned Orbital Docking?

**Comment by peyre** [comment]
If we should implement this in a future version, I think it would be a very helpful thing to have linked to from Future Missions as well as wherever else, and hitting Continue from this could bring you back there if you came from there.  Alternatively, instead of one Continue button it could have a button for Spaceport and another for Future Missions, or what have you.

**Comment by peyre** [comment]
Here's an idea.  Maybe we could put it in the place of Hardware Purchase.  I almost always get to Purchasing by way of R&D.

![Repurpose Hardware Purchase button for Progress Screen](https://github.com/raceintospace/raceintospace/assets/3118014/547e4ed5-9fc4-442e-99b3-d7aad8cb3b18)

**Comment by peyre** [comment]
On second thought, maybe a separate building would be better.  I'll try to come up with artwork for it.  Also, we have to decide what to call it.  Was there a Special Planning building, or something like that, for either side?

**Comment by peyre** [comment]
And we would have to figure out where to put them.  Here are the full spaceports, I think:

![US Spaceport full](https://github.com/raceintospace/raceintospace/assets/3118014/ba0cae20-b842-4184-a4da-b9ba11402b98)
![Soviet spaceport full](https://github.com/raceintospace/raceintospace/assets/3118014/d6c2505f-ff17-4a77-9c42-59b95b2e81e7)

**Comment by peyre** [comment]
Maybe the US building could go in that dark green triangle-like area next to Basic Training, and the Soviet side could squeeze it in above the VIB (mirrored horizontally).

**Comment by peyre** [comment]
You know, that mockup I did of the Progress Summary should probably also accommodate direct ascent, for the odd time that a player takes that route.  Maybe replace the docking module image with a direct-ascent capsule and its stats if the user has prototyped one?

**Comment by peyre** [comment]
If I knew how to do this, I'd build it this weekend, now that 2.0 is out.

**Comment by hweimer** [comment]
The fully equipped ports are a bit larger, especially if you account for currently unimplemented stuff like the Zond building.

![Image](https://github.com/user-attachments/assets/6c0ebe43-fb82-4d36-8046-e4424164b54f)

![Image](https://github.com/user-attachments/assets/344d58ef-d6d9-41e3-badd-70a194869a1e)

**Comment by peyre** [comment]
Oh yes, tricky.  Perhaps smallish buildings could go up at the top near the launch pads, or something.

Is that the Zond building, across the parking lot / plaza from the Soyuz one?

**Comment by hweimer** [comment]
Yes, the Zond building is right next to Soyuz. However, if you click on it, you enter the regular Soyuz building.

**Comment by peyre** [comment]
How'd you get it to show?

Oh, this also brings up the question of how we'd work the building if we implemented LOK/L1/Zond as per #392.  If Voskhod becomes an expanded Vostok, do we eliminate the Voskhod building, or what?  And would the LOK/L1 use the Zond building?

**Comment by hweimer** [comment]
In usa_port.json and sov_port.json, there is a field containing the maximum level of each building. So I set all the fields in Data->P[i].Port to their maximum value (and commented out a few checks whether you already have 'nauts and the like).

There are also a few buildings I have never heard about: "STRATEGIC VISIONS H.Q." (U.S.), "HALLS OF INTERPLAY" and "LENIN'S STATUE" (Soviet).

**Comment by peyre** [comment]
This bit of discussion we really should be having over in the L1 Issue; I'll copy some of this over there.  I guess we would want to allow LOK without Soyuz; I wonder what it would look like.
